PDND Template e-service definition (#755)
italia/eidas-it-wallet-docs: docs/en/e-service-pdnd-template.rst

.. include:: ../common/common_definitions.rst

Template e-Service PDND
=======================

> To enhance the co-design process of APIs, the PDND provides a specialized tool aimed at optimizing the publication and the reuse of e-services. This functionality is comprehensively defined and regulated within the document below:

Reviewer: ```suggestion
The PDND provides a specialized tool that enhances API co-design processes by optimizing e-service publication and reuse. This functionality is thoroughly defined and regulated in the document below.
```

    - "Linee Guida sull'infrastruttura tecnologica della Piattaforma Digitale Nazionale Dati per l'interoperabilità dei sistemi informativi e delle basi di dati" (`PDND`_).

> Central to this functionality is the concept of the template e-service, which serves as a standardized blueprint encapsulating all the necessary technical and descriptive metadata associated with an e-service. Such templates can be created and maintained by any Participants, referred to as API Co-design Managers and these can be either Providers or Consumers within the PDND ecosystem.

Reviewer: ```suggestion
The template e-service concept is central to this functionality, it serves as a standardized blueprint containing all necessary technical and descriptive metadata for an e-service. API Co-design Managers, who can be either Providers or Consumers within the PDND ecosystem, MAY create and maintain these templates.
```

.. include:: ../common/common_definitions.rst

Template e-Service PDND
=======================

> To enhance the co-design process of APIs, the PDND provides a specialized tool aimed at optimizing the publication, sharing, and reuse of e-services. This functionality is comprehensively defined and regulated within the official document:

Reviewer: ```suggestion
To enhance the co-design process of APIs, the PDND provides a specialized tool aimed at optimizing the publication and the reuse of e-services. This functionality is comprehensively defined and regulated within the document below:
```

    - "Linee Guida sull'infrastruttura tecnologica della Piattaforma Digitale Nazionale Dati per l'interoperabilità dei sistemi informativi e delle basi di dati" (`PDND`_).

Central to this functionality is the concept of the template e-service, which serves as a standardized blueprint encapsulating all the necessary technical and descriptive metadata associated with an e-service. Such templates can be created and maintained by any Participants, referred to as API Co-design Managers and these can be either Providers or Consumers within the PDND ecosystem.

Once a template e-service is published, it is accessible through the PDND Template Catalog, a centralized repository that facilitates discovery and reuse. This catalog enables any authorized PDND Participant to browse available templates and instantiate new e-services based on existing designs.

> This mechanism speeds up the development process and encourages teamwork by using shared and reusable components that follow PDND’s standards.
> 

Reviewer: ```suggestion
```

PDND Template e-service definition and guidelines
-------------------------------------------------

The PDND infrastructure supports the lifecycle management of Template E-Services, similar to that of traditional e-services. The lifecycle states include: **Draft**, **Active**, **Suspended**, and **Deprecated**. As with traditional e-services, PDND enforces role-based access control to govern status transitions.

Templates e-service Management
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Template E-Service Creation
"""""""""""""""""""""""""""
Participants are enabled to create Templates E-Service via a guided wizard accessible through the PDND Web GUI (APIs will be available in the future). The creation workflow closely mirrors that of standard e-service creation, with the following distinctions:

    - An additional field identifies the intended recipient of the template.
    - The "Audience" field is omitted.
    - Thresholds are optional and serve as recommendations for Participants implementing the template.

> Participants are prohibited from creating multiple templates with the same name: template names must be unique per participant.. Upon creation, a template is initially set to the Draft state. Templates can then be published to the Template Catalogue, thereby making them accessible to all Participants.

Reviewer: ```suggestion
Participants are prohibited from creating multiple templates with the same name: template names MUST be unique per participant. Upon creation, a template is initially set to the Draft state. Templates can then be published to the Template Catalogue, thereby making them accessible to all Participants.
```
